### PR TITLE
add simple devcontainer setup

### PR DESCRIPTION
**Pull request by leifwalsh**

### CODE REVIEW

File: .devcontainer/devcontainer.json [hunk 1]
```diff
@@ -0,0 +1,38 @@
+// For format details, see https://aka.ms/devcontainer.json. For config options, see the README at:
+// https://github.com/microsoft/vscode-dev-containers/tree/v0.154.0/containers/rust
+{
+	"name": "Rust",
+	"build": {
+		"dockerfile": "Dockerfile"
```
Comment (blinsay) [inline .devcontainer/devcontainer.json:6]: Can this be `image: mcr.microsoft.com/vscode/devcontainers/rust:0-1`? Seems like that would skip the first build.

File: .devcontainer/devcontainer.json [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+		"lldb.executable": "/usr/bin/lldb",
+		// VS Code don't watch files under ./target
+		"files.watcherExclude": {
+			"**/target/**": true
+		}
+	},
+
+	// Add the IDs of extensions you want installed when the container is created.
+	"extensions": [
+		"rust-lang.rust",
```
Comment (blinsay) [inline .devcontainer/devcontainer.json:22]: Rust analyzer? 🙏 

https://marketplace.visualstudio.com/items?itemName=matklad.rust-analyzer

**Review by blinsay — COMMENTED** [review]
This is nice!

File: .devcontainer/devcontainer.json [hunk 3]
```diff
@@ -0,0 +1,38 @@
+// For format details, see https://aka.ms/devcontainer.json. For config options, see the README at:
+// https://github.com/microsoft/vscode-dev-containers/tree/v0.154.0/containers/rust
+{
+	"name": "Rust",
+	"build": {
+		"dockerfile": "Dockerfile"
+	},
+	"runArgs": [ "--cap-add=SYS_PTRACE", "--security-opt", "seccomp=unconfined" ],
```
Comment (geofft) [inline .devcontainer/devcontainer.json:8]: You shouldn't need CAP_SYS_PTRACE with seccomp=unconfined, but maybe it allows you to use an unprivileged lldb to debug an nsncd running as root?

Comment (leifwalsh) [reply .devcontainer/devcontainer.json:8]: This is the default rust config vscode gave me. I didn't add these, but I can remove them if you think we should.

Comment (geofft) [reply .devcontainer/devcontainer.json:8]: lol, okay, leave it and I'll file a bug report against vscode :)

File: .devcontainer/devcontainer.json (hunk 1 above)
Comment (leifwalsh) [reply .devcontainer/devcontainer.json:6]: idk I took the defaults

File: .devcontainer/devcontainer.json (hunk 3 above)
Comment (leifwalsh) [reply .devcontainer/devcontainer.json:8]: https://github.com/microsoft/vscode-remote-try-rust/blob/master/.devcontainer/devcontainer.json#L8

**Pull request merged**